src/crimson/osd/pg.cc

```diff
@@ -736,19 +752,18 @@ seastar::future<Ref<MOSDOpReply>> PG::do_osd_ops(
       *m,
       obc->obs.oi.soid);
     return seastar::make_ready_future<Ref<MOSDOpReply>>(std::move(reply));
-  }, osd_op_errorator::all_same_way([ox = ox.get(),
+  }, crimson::ct_error::object_corrupted::handle([m,
+                                     obc,
+                                     this] () {
+    return rep_repair_primary_object(m, obc->obs.oi.soid, obc->obs.oi.version).then([]() -> PG::do_osd_ops_ertr::future<Ref<MOSDOpReply>> {
+      return crimson::ct_error::eagain::make();
+    });
+  }), OpsExecuter::osd_op_errorator::all_same_way([ox = ox.get(),
                                      m,
                                      obc,
                                      this] (const std::error_code& e) {
     return handle_failed_op(e, std::move(obc), *ox, *m);
-  })).handle_exception_type([ox_deleter = std::move(ox),
```

Comment:
Yeah, this is the legacy we discussed. I would love to drop it but, at least for some time, `handle_exception_type` + `abort()` would be needed. I just sent https://github.com/ceph/ceph/pull/38907 to expose the method. It's needed as we started having an errorated future at this stage (previously plain `seastar::future` was here).